nemo/collections/nlp/models/language_modeling/megatron_gpt_model.py

```diff
@@ -762,6 +767,15 @@ def training_step(self, dataloader_iter):
         if self.initialize_ub:
             self.initialize_ub_func()
 
+        # Reset learning rate
+        if self.if_init_step and self.reset_lr:
+            self._optimizer.param_groups[0]['lr'] = 0.0 if self.cfg.optim.sched.warmup_steps > 0 else self.cfg.optim.lr
```

Comment:
I assume this is the main place to reset lr? I am not super sure it will work for dist optimizer. 
Also what happens if there are two groups?

Reply:
> I assume this is the main place to reset lr? I am not super sure it will work for dist optimizer. Also what happens if there are two groups?

yes, it works with dist optimizer.
yes, for now it only supports dist optimizer since the're going to use distributed one